src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/trait/BitbucketLegacyTraitConverter.java

```diff
@@ -24,14 +24,18 @@ private BitbucketLegacyTraitConverter() {
      */
     @CheckForNull
     public static SCMSourceTrait maybeConvert(SCMSourceTrait trait) {
-        if (trait instanceof TagDiscoveryTrait || trait instanceof DiscoverOtherRefsTrait) {
```

Comment:
So I suppose customers that upgraded to 4.0 before this merges will lose their tag discovery trait and need to create the new one, but those that upgrade after don't need to make any changes? We should communicate this in the release notes and on the bug ticket- this feels like a gotcha.

Reply:
From my understanding they may still need to make changes to their discovery trait as it's a new one. But this is something we can test.

Reply:
I think that's worth doing- maybe as part of the blitz?